src/reanimated2/component/KeyboardAvoidingView.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { forwardRef, useCallback } from 'react';
+import type {
+  KeyboardAvoidingViewProps,
+  LayoutRectangle,
+  View,
+  ViewProps,
+} from 'react-native';
+import { StyleSheet, useWindowDimensions } from 'react-native';
+import { AnimatedView as ReanimatedView } from './View';
+import {
+  useAnimatedKeyboard,
+  useAnimatedStyle,
+  useWorkletCallback,
+  runOnUI,
+  useSharedValue,
+} from '../';
+
+type Props = ViewProps & KeyboardAvoidingViewProps;
+
+/**
+ * View that moves out of the way when the keyboard appears by automatically
+ * adjusting its height, position, or bottom padding.
+ */
+const KeyboardAvoidingView = forwardRef<View, React.PropsWithChildren<Props>>(
+  (
+    {
+      behavior,
+      children,
+      contentContainerStyle,
+      enabled = true,
+      keyboardVerticalOffset = 0,
+      style,
+      onLayout,
+      ...props
+    },
+    ref
+  ) => {
+    const initialFrame = useSharedValue<LayoutRectangle | null>(null);
+
+    const currentFrame = useSharedValue<LayoutRectangle | null>(null);
+
+    const keyboard = useAnimatedKeyboard();
+    const { height: screenHeight } = useWindowDimensions();
+
+    const onLayoutWorklet = useWorkletCallback((layout: LayoutRectangle) => {
+      if (initialFrame.value == null) {
+        initialFrame.value = layout;
+      }
+
+      currentFrame.value = layout;
+    });
+
+    const handleOnLayout = useCallback<NonNullable<ViewProps['onLayout']>>(
+      (event) => {
+        runOnUI(onLayoutWorklet)(event.nativeEvent.layout);
+
+        if (onLayout) {
+          onLayout(event);
+        }
+      },
+      [onLayout]
+    );
+
+    const getBackwardCompatibleBottomHeight = useWorkletCallback(
+      (keyboardHeight: number) => {
+        if (currentFrame.value == null || initialFrame.value == null) {
+          return 0;
+        }
+
+        const keyboardY =
+          screenHeight - keyboardHeight - keyboardVerticalOffset;
+
+        if (behavior === 'height') {
+          return Math.max(initialFrame.value.height - keyboardHeight, 0);
+        }
+
+        return Math.max(
+          currentFrame.value.y + currentFrame.value.height - keyboardY,
+          0
+        );
+      }
+    );
```

Comment:
Same as above, I advise against usage of _useWorkletCallback_ unless it is fixed. This hook also has no dependencies whatsoever - e.g. `behavior` was always _undefined_.
```suggestion
    const getBackwardCompatibleBottomHeight = useCallback(
      (keyboardHeight: number) => {
        'worklet';
        if (currentFrame.value == null || initialFrame.value == null) {
          return 0;
        }

        const keyboardY =
          screenHeight - keyboardHeight - keyboardVerticalOffset;

        if (behavior === 'height') {
          return Math.max(initialFrame.value.height - keyboardHeight, 0);
        }

        return Math.max(
          currentFrame.value.y + currentFrame.value.height - keyboardY,
          0
        );
      },
      [
        behavior,
        currentFrame,
        initialFrame,
        keyboardVerticalOffset,
        screenHeight,
      ]
    );
```